src/pages/index.js

```diff
@@ -35,13 +35,39 @@ const Home = ({ originalLayout, error }) => {
     return clearInterval(useInterval);
   }, []);
 
+  useEffect(() => {
+    if (layout[0].beats.length === beatsMeasure) return;
+    const newLayout = [...layout];
+    validateBeats(newLayout);
+  }, [beatsMeasure]);
+
   const onPlayButtonClick = active => {
     if (!active) {
       setCount(1);
     }
     setPlay(active);
   };
 
+  const validateBeats = layout => {
+    let beatLength = layout[0].beats.length;
+    while (!Number.isInteger(beatLength / beatsMeasure)) {
+      layout.forEach(row => {
+        beatLength < beatsMeasure ? row.beats.push(false) : row.beats.pop();
+      });
+      beatLength = layout[0].beats.length;
+    }
+    return setLayout(layout);
+  };
+
+  const updateBeats = (layout, remove) => {
+    layout.forEach(row => {
```

Comment:
- [ ] Let's use the [filter array function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead here.